aiidalab_widgets_base/viewers.py

```diff
@@ -546,9 +546,10 @@ def apply_selection(self, _=None):
 
     def download(self, change=None):  # pylint: disable=unused-argument
         """Prepare a structure for downloading."""
+        pk = f"pk-{self.pk}" if self.pk else "not-stored"
```

Comment:
I've adjusted this in the comment below. Further, this should probably only be executed when needed.
```suggestion
```